src/Symfony/Contracts/Deprecation/deprecated.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Triggers a deprecation.
+ *
+ * As recommended for prod, turn the "zend.assertions" ini setting to 0 or -1 to disable deprecation notices.
+ *
+ * The function doesn't use type hints nor variadics on purpose, to make it as fast as possibly achievable.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+function deprecated($package, $version, $message)
+{
+    assert(trigger_error("Since $package $version: ".sprintf($message, ...array_slice(func_get_args(), 3)), E_USER_DEPRECATED));
```

Comment:
shouldn't it use a silenced deprecation so that it is not turned into an exception by error handlers ?